backend/src/main/java/com/bakdata/conquery/models/events/CBlock.java

```diff
@@ -307,12 +315,47 @@ public static long calculateBitMask(int pathIndex, int[] mostSpecificChild) {
 					}
 
 					CDateRange range = bucket.getAsDateRange(event, column);
-					spans[index] = spans[index].spanClosed(range);
+
+					if (range.hasLowerBound()) {
+						final int minValue = range.getMinValue();
+
+						maxs[index] = Math.max(maxs[index], minValue);
+						mins[index] = Math.min(mins[index], minValue);
+					}
+
+					if (range.hasUpperBound()) {
+						final int maxValue = range.getMaxValue();
+
+						maxs[index] = Math.max(maxs[index], maxValue);
+						mins[index] = Math.min(mins[index], maxValue);
+					}
 				}
 
 			}
 		}
 
+		CDateRange[] spans = new CDateRange[bucketSize];
+
+		for (int index = 0; index < bucketSize; index++) {
+			if (mins[index] == Integer.MAX_VALUE && maxs[index] == Integer.MIN_VALUE) {
+				spans[index] = CDateRange.all();
+				continue;
+			}
+
+			if (mins[index] == Integer.MAX_VALUE) {
+				spans[index] = CDateRange.atMost(maxs[index]);
+				continue;
+			}
+
+			if (maxs[index] == Integer.MAX_VALUE) {
+				spans[index] = CDateRange.atLeast(mins[index]);
+				continue;
+			}
+
+			spans[index] = CDateRange.of(mins[index], maxs[index]);
+		}
+
+
```

Comment:
Ich wurde den Loop-Body in eine Methode packen und die dann in der oberen äußeren Loop aufrufen ohne dass dann `mins` und `maxs` existieren muss.

Ich weiß dass du hier Phase Splitting machst, aber für mich ist das eins zuweit draußen, da die beiden for-loops doch über die selbe Sache iterieren

Reply:
Dann musst du aber genau den komplexen Teil vom span-closed nachbauen und das will ich hier vermeiden